src/activity_handlers.cpp

```diff
@@ -4392,26 +4392,24 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
 
 static void blood_magic( player *p, int cost )
 {
-    static std::array<bodypart_id, 6> part = { {
-            bodypart_id( "head" ), bodypart_id( "torso" ), bodypart_id( "arm_l" ), bodypart_id( "arm_r" ), bodypart_id( "leg_l" ), bodypart_id( "leg_r" )
-        }
-    };
-    int max_hp_part = 0;
     std::vector<uilist_entry> uile;
-    for( int i = 0; i < num_hp_parts; i++ ) {
-        uilist_entry entry( i, p->hp_cur[i] > cost, i + 49, body_part_hp_bar_ui_text( part[i] ) );
-        if( p->hp_cur[max_hp_part] < p->hp_cur[i] ) {
-            max_hp_part = i;
-        }
-        const std::pair<std::string, nc_color> &hp = get_hp_bar( p->hp_cur[i], p->hp_max[i] );
+    std::vector<bodypart_id> parts;
+    int i = 0;
+    for( const std::pair<bodypart_str_id, bodypart> &part : p->get_body() ) {
```

Comment:
Since `get_body` returns a map, this should iterate be
```suggestion
    for( const std::pair<const bodypart_str_id, bodypart> &part : p->get_body() ) {
```
Applies to all other iterations over the result of `get_body()` (btw. you do this already at some, but not all places).

Reply:
Thanks, I thought that if the pair is const I don't need to specify the id to be const.

Reply:
`std::pair<int, char>` and `std::pair<const int, char>` are completely different and unrelated types. They can be converted, which is what the compiler does.